webapp/channels/src/components/advanced_create_post/advanced_create_post.tsx

```diff
@@ -1040,7 +1040,7 @@ class AdvancedCreatePost extends React.PureComponent<Props, State> {
 
     handleUploadError = (err: string | ServerError, clientId?: string, channelId?: string) => {
         let serverError = err;
-        if (typeof serverError === 'string') {
+        if (typeof serverError === 'string' && serverError.length !== 0) {
```

Comment:
only change is this one.
Rest everything else if converting to functional component for no reason whatsoever :D